spark/src/main/scala/geotrellis/spark/io/avro/codecs/TileCodecs.scala

```diff
@@ -32,7 +35,9 @@ trait TileCodecs {
         .asScala // notice that Avro does not have native support for Short primitive
         .map(_.toShort)
         .toArray
-      new ShortArrayTile(array, rec[Int]("cols"), rec[Int]("rows"))
+      val cellType = CellType.fromString(rec[Utf8]("celltype").toString)
```

Comment:
Since this is stored per record I'm going to pinch bits here. The first part of cell type, the data type is irrelevant because you already know that. We have to specialize avro codecs per data type so we allocate arrays of min size. Looks like what is needed here is some kind of enum for the type of no data, `RAW`, `CONSTANT`, `USER_DEFINED` (with default being `CONSTANT`) and then an optional NoData field. This would also make it easier to read for a hypothetical client that used our avro schema.